### ISSUE TITLE
Exception setting "expires": "Cannot convert null to type "System.DateTimeOffset"."

### ISSUE THREAD

**Issue by GeorgeSchiro** [PKISharp/ACMESharpCore-PowerShell#65]
We started full regression testing of "1.1.5" this morning and noticed an error not seen before. So we went back to "1.1.4" (which tested fine several days ago) to see if we get the same error. We did:

    $challenge | Complete-ACMEChallenge "C:\Users\admin\Desktop\tmp\AcmeState"

    Exception calling "GetResult" with "0" argument(s): "An error occurred while sending the request."
    At C:\Users\admin\Desktop\tmp\ACME-PS\1.1.4\ACME-PS.psm1:1344 char:5
    +     $httpResponse = $httpClient.SendAsync($httpRequest).GetAwaiter(). ...
    +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
        + FullyQualifiedErrorId : HttpRequestException
 
    Exception setting "expires": "Cannot convert null to type "System.DateTimeOffset"."
    At C:\Users\admin\Desktop\tmp\ACME-PS\1.1.4\ACME-PS.psm1:750 char:9
    +         $this.expires = $httpResponse.Content.expires;
    +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        + CategoryInfo          : NotSpecified: (:) [], SetValueInvocationException
        + FullyQualifiedErrorId : ExceptionWhenSetting

At first we thought it might be a problem with LE. Then we bounced the server and the problem went away.

Should this error possibly be handled differently? Perhaps at least a null check on the "expires" value?

**Comment by glatzert** [comment]
Hi George,
that error is possibly existent since day one and might only occur if the challenge is already resolved, which might happen, if you send an order for DNS names you recently ordered.
Is that the case?

Am 13.02.2020 18:35 schrieb George Schiro <notifications@github.com>:

We started full regression testing of "1.1.5" this morning and noticed an error not seen before. So we went back to "1.1.4" (which tested fine several days ago) to see if we get the same error. We did:

$challenge | Complete-ACMEChallenge "C:\Users\admin\Desktop\tmp\AcmeState"

Exception calling "GetResult" with "0" argument(s): "An error occurred while sending the request."
At C:\Users\admin\Desktop\tmp\ACME-PS\1.1.4\ACME-PS.psm1:1344 char:5
+     $httpResponse = $httpClient.SendAsync($httpRequest).GetAwaiter(). ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : HttpRequestException

Exception setting "expires": "Cannot convert null to type "System.DateTimeOffset"."
At C:\Users\admin\Desktop\tmp\ACME-PS\1.1.4\ACME-PS.psm1:750 char:9
+         $this.expires = $httpResponse.Content.expires;
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], SetValueInvocationException
    + FullyQualifiedErrorId : ExceptionWhenSetting


At first we thought it might be a problem with LE. Then we bounced the server and the problem went away.

Should this error possibly be handled differently? Perhaps at least a null check on the "expires" value?

**Comment by GeorgeSchiro** [comment]
Nope, it was definitely a new order with 5 new challenges. We ran exactly the same test (ie. with new order with new challenges) many times and got the same error (not every time) on different challenges (sometimes the same one, but usually not). In fact, a similar error happened once during the order phase (ie. before the first challenge).

We bounced the server, did exactly the same thing again (several times) and the error did not recur. It looks like a network issue to me (ie. something wrong with the IP stack on our server).

As usual, I was just hoping for something more useful from the error feedback. Not that big a deal though Thomas.

**Comment by glatzert** [comment]
Sorry - was in Bus and did not read well enough...

The error happens in `Invoke-AcmeWebRequest` which does handle Responses in general, but not the *no-response* case.

If you want to try a fix, then this code will help:
```
try {
        $httpClient = [System.Net.Http.HttpClient]::new();
        $httpResponse = $httpClient.SendAsync($httpRequest).GetAwaiter().GetResult();
    } catch {
        $result = [AcmeHttpResponse]::new();
        $result.IsError = $true;
        $result.Message = $_.Message;
        $result.Content = $_;
    }
```

if not, then just wait for next week, where I can test the code in my environment and I'll do 1.1.6

**Comment by GeorgeSchiro** [comment]
Sadly, I can't reproduce the error anymore Thomas.

**Comment by glatzert** [comment]
Heh.. pulling the network cord will make it reproduce :-D

**Comment by GeorgeSchiro** [comment]
Ok, hold on ... Shit - for some reason the help desk phone bank just lit up ...

**Comment by glatzert** [comment]
I'm closing this, since it will be finally fixed in 1.1.6 (along with other small glitches)
